salt/minion.py

```diff
@@ -3165,6 +3165,28 @@ def _post_master_init(self, master):
             self.schedule.delete_job(master_event(type='alive', master=self.opts['master']), persist=True)
             self.schedule.delete_job(master_event(type='failback'), persist=True)
 
+        # proxy keepalive
+        proxy_alive_fn = fq_proxyname+'.alive'
+        if proxy_alive_fn in self.proxy and 'status.proxy_reconnect' in self.functions and \
+           ('proxy_keep_alive' not in self.opts or ('proxy_keep_alive' in self.opts and self.opts['proxy_keep_alive'])):
+            # if `proxy_keep_alive` is either not specified, either set to False does not retry reconnecting
+            self.schedule.add_job({
+                '__proxy_keepalive':
+                {
+                    'function': 'status.proxy_reconnect',
+                    'minutes': self.opts.get('proxy_keep_alive_interval', 1),  # by default, check once per minute
+                    'jid_include': True,
+                    'maxrunning': 1,
+                    'return_job': False,
+                    'kwargs': {
+                        'proxy_name': fq_proxyname,
+                        'opts': self.opts
```

Comment:
The reasoning I'm passing this is for reopening the connection (see https://github.com/cloudflare/salt/blob/bfabbcaff721110b65b33251c42133836d0382a4/salt/modules/status.py#L1136-L1137):

```python
    is_alive = __proxy__[proxy_keepalive_fn](opts)
    if not is_alive:
        __proxy__[proxy_name+'.shutdown'](opts)  # safely close connection
        __proxy__[proxy_name+'.init'](opts)  # reopen connection
```

As ```opts``` need to be sent to ```init``` and ```shutdown```. If that's a blocker, perhaps we should find a different function? Or simply use the ```__opts__``` object inside the ```status``` module?